internal/access/grant.go

```diff
@@ -55,14 +55,19 @@ func ListGrants(rCtx RequestContext, opts data.ListGrantsOptions, lastUpdateInde
 		return ListGrantsResponse{Grants: result}, err
 	}
 
+	dest, err := data.GetDestination(rCtx.DBTxn, data.GetDestinationOptions{ByName: opts.ByDestination})
```

Comment:
It looks like `ByDestination` is optional in `ListGrants` calls. Will this error when `ByDestination` isn't specified? In general it feels safer to skip this look-up if `ByDestination` isn't set.

Reply:
Good call out!

It's not super obvious from here, but we have to be very strict about which parameters are used with long polling. We only support long polling with a `destination` query parameter, and no other query parameters. That's implemented here:

https://github.com/infrahq/infra/blob/64d306a5ae388ebd3cb272b91bd6263cb69dd93b/api/grant.go#L73-L91

On line 53 above we check if `lastUpdateIndex` is 0, so the only way to get to line 58 here is for long polling. In the future if we have other supported parameter combinations we'll have to make sure this destination lookup is only done for the destination parameter.

All the code below this also requires the destination parameter. So our check for "can this continue" is line 53. I don't think it would make sense to skip this lookup here, because the rest of the code would also need to be skipped.

We could add another `if opts.ByDestination == "" { ... } ` that returns and comment that it's an extra safeguard that is normally handled by request validation.

Reply:
I think the request validation seems ok for now, thanks for the clarification.